src/layer/__tests__/cache.ts

```diff
@@ -58,9 +68,26 @@ describe('Testing caching', () => {
     mockNetwork.onPost().replyOnce(200, mockedResponse);
     mockNetwork.onPost().replyOnce(200, mockedResponse);
 
-    await layer.findTiles(bbox, fromTime, toTime, null, null, requestsConfig);
+    const responseFromMockNetwork = await layer.findTiles(bbox, fromTime, toTime, null, null, requestsConfig);
+    const fromCacheResponse = await layer.findTiles(bbox, fromTime, toTime, null, null, requestsConfig);
+    expect(fromCacheResponse.tiles).toStrictEqual(expectedResultTiles);
+    expect(fromCacheResponse.hasMore).toBe(expectedResultHasMore);
+    expect(fromCacheResponse).toStrictEqual(responseFromMockNetwork);
+
     await new Promise(r => setTimeout(r, 1100));
-    await layer.findTiles(bbox, fromTime, toTime, null, null, requestsConfig);
+
+    const responseFromMockNetwork2 = await layer.findTiles(
+      bbox,
+      fromTime,
+      toTime,
+      null,
+      null,
+      requestsConfig,
+    );
+    const fromCacheResponse2 = await layer.findTiles(bbox, fromTime, toTime, null, null, requestsConfig);
+    expect(fromCacheResponse2.tiles).toStrictEqual(expectedResultTiles);
+    expect(fromCacheResponse2.hasMore).toBe(expectedResultHasMore);
+    expect(fromCacheResponse2).toStrictEqual(responseFromMockNetwork2);
 
     expect(mockNetwork.history.post.length).toBe(2);
```

Comment:
I would suggest being more explicit about which method calls are using network and which are not. In other words, I think we should change these lines to:

```javascript
    const responseFromMockNetwork = await layer.findTiles(bbox, fromTime, toTime, null, null, requestsConfig);
    expect(mockNetwork.history.post.length).toBe(1);

    const fromCacheResponse = await layer.findTiles(bbox, fromTime, toTime, null, null, requestsConfig);
    expect(mockNetwork.history.post.length).toBe(1); // no network request - cache was used
    expect(fromCacheResponse.tiles).toStrictEqual(expectedResultTiles);
    expect(fromCacheResponse.hasMore).toBe(expectedResultHasMore);
    expect(fromCacheResponse).toStrictEqual(responseFromMockNetwork);

    await new Promise(r => setTimeout(r, 1100));

    const responseFromMockNetwork2 = await layer.findTiles(
      bbox,
      fromTime,
      toTime,
      null,
      null,
      requestsConfig,
    );
    expect(mockNetwork.history.post.length).toBe(2);

    const fromCacheResponse2 = await layer.findTiles(bbox, fromTime, toTime, null, null, requestsConfig);
    expect(mockNetwork.history.post.length).toBe(2); // no network request - cache was used
    expect(fromCacheResponse2.tiles).toStrictEqual(expectedResultTiles);
    expect(fromCacheResponse2.hasMore).toBe(expectedResultHasMore);
    expect(fromCacheResponse2).toStrictEqual(responseFromMockNetwork2);
```

Reply:
Changed to what to you suggested